src/main/java/com/ibm/watson/litelinks/SSLHelper.java

```diff
@@ -385,32 +389,40 @@ private static KeyManagerFactory getKeyManagerFactory(KeyStore keyStore, char[]
     }
 
     private static SslContextBuilder configureTrustManager(SslContextBuilder scb, KeyStore trustStore,
-            String trustMgrAlg, File trustCertsFile) throws IOException, GeneralSecurityException {
+            String trustMgrAlg, File trustCertsFile, boolean reqClientAuth) throws IOException, GeneralSecurityException {
         if (TRUST_EVERYTHING) {
             return scb.trustManager(InsecureTrustManagerFactory.INSTANCE);
         }
-        // trust certs file or dir but no truststore
-        if (trustCertsFile != null && trustStore == null) {
-            return !trustCertsFile.isDirectory() ? scb.trustManager(trustCertsFile)
-                    : scb.trustManager(generateCertificates(trustCertsFile).toArray(new X509Certificate[0]));
-        }
-        // all other cases
         TrustManagerFactory tmf = TrustManagerFactory.getInstance(trustMgrAlg);
-        if (trustStore != null) {
-            // truststore *and* certs file or dir provided, add certs to truststore
-            if (trustCertsFile != null) {
-                final String prefix = trustCertsFile.getName();
-                int index = 0;
-                for (X509Certificate cert : generateCertificates(trustCertsFile)) {
-                    String alias;
-                    do {
-                        alias = prefix + '_' + index++;
-                    } while (trustStore.isCertificateEntry(alias));
-                    trustStore.setCertificateEntry(alias, cert);
-                }
+        // Add any provided certs to truststore
+        if (trustCertsFile != null) {
+            if (trustStore == null) {
+                trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
+                trustStore.load(null, null);
+            }
+            final String prefix = trustCertsFile.getName();
+            int index = 0;
+            for (X509Certificate cert : generateCertificates(trustCertsFile)) {
+                String alias;
+                do {
+                    alias = prefix + '_' + index++;
+                } while (trustStore.isCertificateEntry(alias));
+                trustStore.setCertificateEntry(alias, cert);
             }
         }
         tmf.init(trustStore); // passing null here will init with java defaults
+      
+        // Wrap X.509 TrustManager in one which allows use of non-CA certs in client authentication
+        for (TrustManager tm : tmf.getTrustManagers()) {
+            if (tm instanceof X509TrustManager) {
+                if(reqClientAuth)
+                {
```

Comment:
This `if` can go on the outside, i.e. `\\ Wrap X.509...` comment inside it. Also please keep opening bracket on same line for style consistency.